draft-irtf-cfrg-aegis-aead.md

```diff
@@ -792,12 +792,12 @@ The `Decrypt` function decrypts a ciphertext, verifies that the authentication t
 
 Security:
 
-- If tag verification fails, the decrypted message and wrong message authentication tag MUST NOT be given as output. The decrypted message MUST be overwritten with zeros before being returned.
+- If tag verification fails, the decrypted message and wrong authentication tag MUST NOT be given as output. The decrypted message MUST be overwritten with zeros before the function returns.
 - The comparison of the input `tag` with the `expected_tag` MUST be done in constant time.
 
 Inputs:
 
-- `ct`: the ciphertext to be decrypted (length MUST be less than or equal to `C_MAX`).
+- `ct`: the ciphertext to decrypt (length MUST be less than or equal to `C_MAX`).
 - `tag`: the authentication tag.
 - `ad`: the associated data to authenticate (length MUST be less than or equal to `A_MAX`).
 - `key`: the encryption key.
```

Comment:
See higher up comment about the tag/key/nonce lengths.

Reply:
The algorithms are impossible to implement with a different tag/key/nonce length.

Whereas the explicit mention of the limits for `ct`, `msg` and `ad` suggests that implementers should check for this at runtime, or use appropriate types.